resources/sts/4.12/hypershift/InstallerPolicy.json

```
@@ -4,67 +4,130 @@
-                "ec2:CreateSubnet",
-                "ec2:CreateVolume",
-                "ec2:CreateVpc",
-                "ec2:DeleteSubnet",
-                "ec2:DeleteVolume",
-                "ec2:DeleteVpc",
-                "ec2:DescribeAddresses",
+                "ec2:CreateTags",
+                "ec2:DeleteTags",
                 "ec2:DescribeAvailabilityZones",
                 "ec2:DescribeInternetGateways",
-                "ec2:DescribeNetworkInterfaces",
                 "ec2:DescribeRegions",
                 "ec2:DescribeReservedInstancesOfferings",
                 "ec2:DescribeRouteTables",
-                "ec2:DescribeSecurityGroups",
                 "ec2:DescribeSubnets",
                 "ec2:DescribeVpcAttribute",
                 "ec2:DescribeVpcs",
-                "ec2:RunInstances",
-                "elasticloadbalancing:CreateLoadBalancer",
+                "ec2:DescribeInstanceTypeOfferings",
                 "elasticloadbalancing:DeleteLoadBalancer",
-                "iam:AddRoleToInstanceProfile",
-                "iam:AttachRolePolicy",
-                "iam:CreateInstanceProfile",
-                "iam:CreateOpenIDConnectProvider",
-                "iam:CreatePolicy",
-                "iam:CreatePolicyVersion",
-                "iam:CreateRole",
-                "iam:DeleteOpenIDConnectProvider",
-                "iam:DeletePolicyVersion",
-                "iam:DeleteRole",
-                "iam:DeleteRolePermissionsBoundary",
-                "iam:DetachRolePolicy",
-                "iam:GetOpenIDConnectProvider",
-                "iam:GetPolicy",
-                "iam:GetRole",
-                "iam:GetUser",
-                "iam:ListAttachedRolePolicies",
-                "iam:ListPolicyTags",
-                "iam:ListPolicyVersions",
-                "iam:ListRoleTags",
-                "iam:PutRolePermissionsBoundary",
-                "iam:TagPolicy",
                 "kms:DescribeKey",
-                "route53:ChangeResourceRecordSets",
-                "route53:DeleteHostedZone",
-                "route53:GetHostedZone",
+                "iam:GetRole",
                 "route53:ListHostedZones",
-                "route53:ListHostedZonesByName",
                 "route53:ListResourceRecordSets",
-                "s3:CreateBucket",
-                "s3:DeleteBucket",
-                "s3:DeleteObject",
-                "s3:ListObjects",
-                "s3:PutObject",
-                "secretsmanager:GetSecretValue",
                 "servicequotas:GetServiceQuota",
-                "sts:AssumeRole",
-                "sts:AssumeRoleWithWebIdentity",
                 "sts:GetCallerIdentity"
             ],
-        "Resource": "*"
+            "Resource": "*"
+        },
+        {
+            "Action": [
+                "iam:PassRole"
+            ],
+            "Resource": [
+                "arn:*:iam::*:role/*-Worker-Role"
+            ],
+            "Effect": "Allow"
+        },
```

```suggestion
        {
            "Action": [
                "iam:PassRole"
            ],
            "Resource": [
                "arn:*:iam::*:role/*-Worker-Role"
            ],
            "Effect": "Allow",
            "Condition": {
                "StringEquals": {
                    "iam:PassedToService": [
                        "ec2.amazonaws.com"
                    ]
                }
            }
        },
```

Further restriction ensuring the role can only passed the AWS EC2 service.

---

Thanks fixed